### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by sertoa812** [OpenSSE/opensse-schemes#15]
I have following the suggested procedures to install the dependency,
but I have come to the error in opensee-schemes make procedure,

the error is shown as follows,
```
[ 37%] Built target sse_crypto
[ 41%] Built target dbparser
[ 54%] Built target schemes
[ 68%] Built target runners
[ 70%] Linking CXX executable sophos_client
../lib/librunners.so: undefined reference to `grpc_core::JoinHostPort(absl::lts_2020_02_25::string_view, int)'
```

Is there any thing wrong with the grpc installation? 
But I have successfully installed grpc with cmake and make. 
The only reason I suspect is that should I cmake grpc with option shared libraries (.so)?

**Comment by tanhunggg** [comment]
I have the same error ,but I already cmake grpc with option shared libraries , if you solve the error, pls tell me, thank you very much!

**Comment by rbost** [comment]
It seems gRPC sightly changed in the way it is built and now embeds a proper and clean CMake module. I will check if changing that solves the problem, and if yes, update the repo.

**Comment by tanhunggg** [comment]
> It seems gRPC sightly changed in the way it is built and now embeds a proper and clean CMake module. I will check if changing that solves the problem, and if yes, update the repo.

If it is convenient, can you tell me the version of gRPC used to implement the lates version of the code?

**Comment by rbost** [comment]
Hi, and sorry for the late heads up.

I was able to fix the compilation both on a personal computer running Ubuntu 18, and on Travis CI running Xenial (Ubuntu 16). The build and tests pass using gcc (see [this Travis job](https://travis-ci.org/github/OpenSSE/opensse-schemes/jobs/755939053)) and most probably with clang too ([the build](https://travis-ci.org/github/OpenSSE/opensse-schemes/jobs/755939054) is still running right now).

There still is one issue, not encountered in the CI build: the CMake script I use to find RocksDB would not declare the correct library dependencies when linking against the static version of `librocksdb`. Hence make sure to use the shared library until I find a way to fix that issue.

The commit fixing all these issues is 02cb66a8f79c2a77db2625bef6fb520917d89d16. You will need to run `git submodule update` before recompiling everything, as I also had to fix some of the dependencies in the process.

I will try to properly release these fixes in the `master` branch soon, but I want to fix some stuff coming from the updated static analysis.
